pkg/cluster/registry/harbor/v2/mockserver/server.go

```diff
@@ -0,0 +1,146 @@
+package mockserver
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type HarborProject struct {
+	ID           int
+	Name         string
+	Public       bool
+	Members      []*ProjectMember
+	Repositories []*ProjectRepository
+}
+
+type ProjectMember struct {
+	Username string
+	Role     int
+}
+
+type ProjectRepository struct {
+	Name string
+	Tags []string
+}
+
+type HarborServer struct {
+	R         *mux.Router
+	Projects  map[string]*HarborProject
+	projectID int
+}
+
+func NewHarborServer() *HarborServer {
+	r := mux.NewRouter()
+	s := &HarborServer{
+		R:         r,
+		Projects:  map[string]*HarborProject{},
+		projectID: 1,
+	}
+	r.Path("/api/v2.0/projects/{project}/repositories/{repository}").
+		Methods(http.MethodDelete).HandlerFunc(s.DeleteRepository)
+	return s
+}
+
+func (s *HarborServer) CreateProject(projectName string, metadata map[string]string) {
```

Comment:
Are CreateProject and PushImage unused now?

Reply:
> 

Those two interfaces are not needed as a web server, but for testing, they are reserved as methods